pkg/cri/resource-manager/policy/builtin/topology-aware/flags.go

```diff
@@ -55,16 +61,22 @@ func (fh *fakehints) merge(hints fakehints) {
 // defaultOptions returns a new options instance, all initialized to defaults.
 func defaultOptions() interface{} {
 	return &options{
-		PinCPU:         true,
-		PinMemory:      true,
-		PreferIsolated: true,
-		PreferShared:   false,
-		FakeHints:      make(fakehints),
+		PinCPU:                           true,
+		PinMemory:                        true,
+		PreferIsolated:                   true,
+		PreferShared:                     false,
+		FakeHints:                        make(fakehints),
+		AdditionalReservedPoolNamespaces: make([]string, 0),
 	}
 }
 
 // Register us for configuration handling.
 func init() {
 	config.Register(PolicyPath, PolicyDescription, opt, defaultOptions)
 	config.Register(AliasPath, PolicyDescription, aliasOpt, defaultOptions)
+
+	for _, str := range opt.AdditionalReservedPoolNamespaces {
+		additionalReservedPoolNamespacesRe = append(additionalReservedPoolNamespacesRe,
+			glob.MustCompile(str))
```

Comment:
I think we need to delay this and trigger it from `configNotify()`.

As an additional nano-nit, since we do correctly test-compile glob-patterns, we might as well collect them into a precompiled `policy.reservedPoolNamespaces []filepath.Glob`. And if we choose do that, we could always make sure that slice has `kube-system` as the first item and change the current special casing of `kube-system` to just loop through that `Glob` slice trying to `Match()` each configured item.